R/class-plotdata-line.R

```diff
@@ -0,0 +1,300 @@
+                                              'dataType' = NULL,
+                                              'dataShape' = NULL,
+                                              'displayLabel' = NULL),
+                         facetVariable1 = list('variableId' = NULL,
+                                              'entityId' = NULL,
+                                              'dataType' = NULL,
+                                              'dataShape' = NULL,
+                                              'displayLabel' = NULL),
+                         facetVariable2 = list('variableId' = NULL,
+                                              'entityId' = NULL,
+                                              'dataType' = NULL,
+                                              'dataShape' = NULL,
+                                              'displayLabel' = NULL),
+                         value = character(),
+                         evilMode = logical(),
+                         listVarDetails = list('inferredVariable' = NULL,
+                                               'inferredVarPlotRef' = NULL,
+                                               'listVarPlotRef' = NULL,
+                                               'listVarDisplayLabel' = NULL),
+                         verbose = logical(),
+                         ...,
+                         class = character()) {
+
+  .pd <- newPlotdata(.dt = .dt,
+                     xAxisVariable = xAxisVariable,
+                     yAxisVariable = yAxisVariable,
+                     overlayVariable = overlayVariable,
+                     facetVariable1 = facetVariable1,
+                     facetVariable2 = facetVariable2,
+                     evilMode = evilMode,
+                     listVarDetails = listVarDetails,
+                     verbose = verbose,
+                     class = "lineplot")
+
+  attr <- attributes(.pd)
+
+  x <- toColNameOrNull(attr$xAxisVariable)
+  y <- toColNameOrNull(attr$yAxisVariable)
+  group <- toColNameOrNull(attr$overlayVariable)
+  panel <- findPanelColName(attr$facetVariable1, attr$facetVariable2)
+
+  if (value == 'mean') {
+    
+    mean <- groupMean(.pd, x, y, group, panel)
+    data.table::setnames(mean, c(group, panel, 'seriesX', 'seriesY'))
+    .pd <- mean
+    logWithTime('Mean calculated per X-axis value.', verbose)
+
+  } else if (value == 'median') {
+
+     median <- groupMedian(.pd, x, y, group, panel)
+     data.table::setnames(median, c(group, panel, 'seriesX', 'seriesY'))
+    .pd <- median
+    logWithTime('Median calculated per X-axis value.', verbose)
+
+  }
+  if (attr$xAxisVariable$dataType == 'DATE') {
+    .pd$seriesX <- lapply(.pd$seriesX, format, '%Y-%m-%d')
+  } else { 
+    .pd$seriesX <- lapply(.pd$seriesX, as.character)
+  }
+  if (attr$yAxisVariable$dataType == 'DATE') {
+    .pd$seriesY <- lapply(.pd$seriesY, format, '%Y-%m-%d')
+  } else {
+    .pd$seriesY <- lapply(.pd$seriesY, as.character)
```

Comment:
Why do we want to convert numeric to character for the y axis? Is the client converting back to numbers?

Reply:
i did it bc the constraints in the data service claim to be able to handle dates currently. and r can technically get means and medians for dates. idk that to be particularly useful, but i dont know its really wrong either.

Reply:
oh i see it now. Cool

Reply:
and i see scatter is the same. Makes sense now thank you!